src/basic/ether-addr-util.c

```diff
@@ -150,7 +150,7 @@ static int parse_hw_addr_one_field(const char **s, char sep, size_t len, uint8_t
                 p++;
         }
 
-        if (*p != '\0' && *p != sep)
+        if (sep != 0 && *p != '\0' && *p != sep)
```

Comment:
I'd like to use `'\0'` rather than `0` for consistency with the condition for `*p`.